### ISSUE TITLE
bug: pluck() fails with has_many() assoc thru another model

### ISSUE THREAD

**Issue by thefliik** [neo4jrb/neo4j#1425]
Tested in a new, blank app. Given the following models:

```
class Form 
  include Neo4j::ActiveNode

  has_many :out, :owner_edgs, type: :OWNER, model_class: "Owner"

  def owners
    owner_edgs.to_node
  end
end

class Owner 
  include Neo4j::ActiveNode

  has_one :in, :from_node, type: :OWNER, model_class: "Form"
  has_one :out, :to_node, type: :OWNER, model_class: ["Person", "Organization"]
end

class Person 
  include Neo4j::ActiveNode

  has_many :in, :owner_edgs, type: :OWNER, model_class: "Owner"
end
```

# Expected Result

```
form = Form.create
owner = Owner.create
person = Person.create
owner.from_node = form
owner.to_node = person

form.owners.pluck(:id)
#=> ["uidaf-olkcjn-aeklnc-dadf;lcea"] # pretend its a uuid
```

# Actual Result

```
form = Form.create
owner = Owner.create
person = Person.create
owner.from_node = form
owner.to_node = person

form.owners.pluck(:id)
#=> NoMethodError: undefined method `id_property_name' for nil:NilClass
```

# Additional Information

The query works normally if `class Owner` `has_one :out, :to_node, type: :OWNER, model_class: "Person"`

I can also work around the issue by calling `form.owners.to_a.pluck(:id)`

### Runtime information:

Neo4j database version: 3.1.5
`neo4j` gem version: 8.2.4
`neo4j-core` gem version: 7.2.3

**Comment by thefliik** [comment]
I feel like I've found several issues that involve an association `C` through an association `B` where the association `B` is defined with an array `model_class`. Perhaps they are related.

**Comment by cheerfulstoic** [comment]
The two issues might be related, but probably there's a larger issue.  When I try this I get the error here:

https://github.com/neo4jrb/neo4j/blob/master/lib/neo4j/active_node/query/query_proxy_link.rb#L118

That's there because when you traverse to an association which has an array for `model_class`, the `model` method for the `QueryProxy` is `nil` (basically because it doesn't know which to choose).  I think there are probably a few places where there is an assumption of a `model`.

Check out this PR with a fix:
https://github.com/neo4jrb/neo4j/pull/1426

You should be able to apply it in your `Gemfile` to try it out like so:

```ruby
gem 'neo4j', github: 'neo4jrb/neo4j', branch: '1425_fix'
```

If that works, I can throw a spec around it and release a patch

**Comment by thefliik** [comment]
Fixed by 8.2.5 (see comment at bottom of #1426)
